### ISSUE TITLE
Papi ajax - Permalinks settings

### ISSUE THREAD

**Issue by nlemoine** [wp-papi/papi#218]
### Description

When using a `repeater` with a group a fields, the "add new" button is broken. 

The ajax URL called when hitting the button is returning a 404. I found out the URL is only working when permalinks are enabled. 

This should work in any case. 

Checking the source code, I also noticed you didn't take advantage of the [WordPress ajax "api"](https://codex.wordpress.org/AJAX_in_Plugins), this would probably save some bugs (this one included) and provide a "native" way to achieve all the papi ajax stuffs.

There's also a tiny problem with adding your own endpoints with [`add_rewrite_rule`](https://github.com/wp-papi/papi/blob/e51fd4a9e67e414a53aa5249cc5ddfa08c9eafa9/src/admin/class-papi-admin-ajax.php#L27), you need to flush rewrite rules so they can be added to all rewrite rules. 

### Steps to reproduce

- Set permalinks to "simple" (?p=123)
- Add a repeater property with some properties
- Click on "add new"

### What versions of softwares are you using?

- Browser: Chrome 56
- Papi: 3.1.18
- WordPress: 4.7.2

**Comment by frozzare** [comment]
You get a lot of overhead with WP's built in ajax api so that's why I'm not using it and that will not change.

Simple permalinks should work but haven't been tested since I never use simple permalinks. Will look at a solution so simple permalinks should work.

**Comment by frozzare** [comment]
The steps provided to reproduce this bug isn't working. Setting permalinks to simple doesn't work, I have also tried to reinstall my WordPress installation and reproduce the same steps as you provided without success.

**Comment by nlemoine** [comment]
Sorry, the english name for the permalinks setting is "Plain", not "Simple". 

I can easily reproduce it on a fresh install with this sample type: 
```php
<?php

class AjaxType extends Papi_Page_Type {

	/**
	 * Meta data
	 *
	 * @return array
	 */
	public function meta() {
		return [
			'name'        => 'Name',
			'description' => 'Description',
		];
	}

	/**
	 * Define our properties.
	 */
	public function register() {

		$this->box('Repeater', [
			papi_property([
				'title'    => 'Title',
				'slug'     => 'repeater_property',
				'type'     => 'repeater',
				'settings' => [
					'items' => [
						papi_property([
							'type'     => 'string',
							'title'    => 'Description',
							'slug'     => 'description',
						]),
					]
				]
			] ),
		]);

	}
}

```

**Comment by frozzare** [comment]
1. On a fresh installation I have selected "plain" permalinks
2. Have a repeater property with a child properties
3. `papi-ajax` endpoint works for me, maybe some type of cache in WordPress that isn't flushed or something like that.

Running wp cli don't help.

→ wp rewrite flush
Warning: Rewrite rules are empty, possibly because of a missing permalink_structure option. Use 'wp rewrite list' to verify, or 'wp rewrite structure' to update permalink_structure.

→ wp rewrite list
Warning: No rewrite rules.

So I'm unable to reproduce this right now but I have created a example that don't use `papi-ajax` endpoint when permalink structure is empty, would be great if you can try it https://github.com/wp-papi/papi/commit/126f84210b20bf021f94ee505ff6f901e744f0d0

With this solution we could remove `papi-ajax` endpoint and add a query string `papi-ajax=true` instead so plain permalink structure and the others use the same solution. `papi-ajax` endpoint it's nice but not necessary to use a real endpoint for that.

**Comment by nlemoine** [comment]
I think I know what's the issue here. Are you using Apache? Do the `.htaccess` file contains the WP rewrite rules? If so, you will not see the issue described here. 

Same goes is you're using nginx (rewrite rules are enabled by default). 

In order to reproduce the issue, you must either not have an `.htaccess` file or it must not contain rewrite rules. 

I'll try your fix but I looks like the one I used to fix this. I'll give you some feedback within the day.

**Comment by nlemoine** [comment]
👍
